deprecated.tf

```
@@ -7,6 +7,8 @@ locals {
   cloudfront_access_log_include_cookies = var.log_include_cookies == null ? var.cloudfront_access_log_include_cookies : var.log_include_cookies
   cloudfront_access_log_prefix          = var.log_prefix == null ? var.cloudfront_access_log_prefix : var.log_prefix
 
+  deployment_principals = var.deployment_principal_arns == null ? var.deployment_principals : { for arn, path_prefix in var.deployment_principal_arns : arn => { "arn" : arn, "path_prefix" : path_prefix } }
```

This logic returns the following error in our tests

```
        	            	Error: Inconsistent conditional result types
        	            	
        	            	  on ../../deprecated.tf line 10, in locals:
        	            	  10:   deployment_principals = var.deployment_principal_arns == null ? var.deployment_principals : { for arn, path_prefix in var.deployment_principal_arns : arn => { "arn" : arn, "path_prefix" : path_prefix } }
```

Could you format this to make the logic easier to read? Id also put the second portion of the ternary in a separate local.

```suggestion
  deployment_principal_arns = {
    for arn, path_prefix in var.deployment_principal_arns :
    arn => { "arn" : arn, "path_prefix" : path_prefix }
  }
  deployment_principals = var.deployment_principal_arns == null ? var.deployment_principals : local.deployment_principal_arns
```

Id also try to run the examples/complete module locally to make sure it works as expected. It will be faster iteration.

---

sorry for the delay here.. @nitrocode I've pushed a patch that fixes the readability. I'll report back on running the `examples/complete` module.

---

..hmm we don't use route 53, so I don't have an R53 zones setup. Is there another way to run some tests?

---

@abeluck I re-ran the Terratest here and the result is the same as above

```
│ Error: Inconsistent conditional result types
│ 
│   on ../../deprecated.tf line 17, in locals:
│   17:   deployment_principals = var.deployment_principal_arns == null ? var.deployment_principals : local.deployment_principals_from_deprecated_deployment_principal_arns
│     ├────────────────
│     │ local.deployment_principals_from_deprecated_deployment_principal_arns is object with 2 attributes
│     │ var.deployment_principal_arns is map of list of string with 2 elements
│     │ var.deployment_principals is empty map of object
│ 
│ The true and false result expressions must have consistent types. The
│ 'true' value is map of object, but the 'false' value is object.
╵}
```

I believe the issues is because the [example](https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/blob/main/examples/complete/deployment.tf#L12) expects the map value to be a list of strings, whereas your variable is expecting a single prefix per arn.

---

Thanks @joe-niland It was an easy fix. I pared down the complete test to be able to run it locally. It's working now both with the deprecated `var.deployment_principal_arns` and new `var.deployment_principals`.

I've updated the documentation in a few more places, and rebased the branch so the PR can be easily squash merged.

We could also change the complete example to use the new `var.deployment_principals`, what do you think?

---

Thanks @abeluck, that sounds good.
Yes, I think it makes sense to update the example.

---

@abeluck sorry for the delay - all looks good, but will you be updating the examples?

---

@abeluck see comment above re: examples :)